common/persistence/statsComputer.go

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package persistence
+
+type (
+	// statsComputer is to computing struct sizes after serialization
+	statsComputer struct{}
+)
+
+func (sc *statsComputer) computeMutableStateStats(req *PersistenceUpdateWorkflowExecutionRequest) *MutableStateStats {
+	executionInfoSize := computeExecutionInfoSize(req.ExecutionInfo)
+
+	activityInfoCount := 0
+	activityInfoSize := 0
+	for _, ai := range req.UpsertActivityInfos {
```

Comment:
This is not going to work.  Mutable state stats cannot be computed using update request.  It only has the upsert activity/timer/child/signal infos, not the entire size of mutable state.